### PR TITLE
turn on server sync for gun light component

### LINKED ISSUE #20247: crusher flashlight turns off when shooting
## Description
should either remove the flashlight, the shooting light effect, or make it support both at the same time

**Reproduction**
1. get crusher
2. use light action
3. wield
4. shoot
5. light off

**Screenshots**

https://github.com/space-wizards/space-station-14/assets/39013340/f5b33085-abb8-43f3-8a93-a73415b1bc24

### PR DESCRIPTION
**Pull request by Costyyy**
## About the PR

Turned on net sync for the gun light component as it seems that the muzzle flash animation overrides the light from the
flashlight and doesn't bring it back to the original state.
## Why / Balance

Fix #20247 
## Media

https://github.com/space-wizards/space-station-14/assets/48994982/fda7249f-8de3-483f-a801-91ec724a9235

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Crusher flashlight light no longer disappears after shooting.

### CODE REVIEW

**Comment by deltanedas** [comment]
i think sloth said something about light layers

**Comment by Costyyy** [comment]
I'm not sure about the performance impact of this change. I had an idea of enabling net sync only if the light component's Enabled member is set to true, which as I understand would mean that a light is turned on. But it seems that Enabled remains on true permanently even if the light component isn't actually emitting light anymore.

**Comment by Costyyy** [comment]
> i think sloth said something about light layers

Is that a system that exists at the moment or is the idea to implement such a system?

File: Content.Client/Weapons/Ranged/Systems/GunSystem.cs [hunk 1]
```diff
@@ -310,7 +310,7 @@ protected override void CreateEffect(EntityUid uid, MuzzleFlashEvent message, En
 
         _animPlayer.Play(ent, anim, "muzzle-flash");
         var light = EnsureComp<PointLightComponent>(uid);
-        light.NetSyncEnabled = false;
+        light.NetSyncEnabled = true;
```
Comment (metalgearsloth) [inline Content.Client/Weapons/Ranged/Systems/GunSystem.cs:313]: Might be better to just only disable netsyncing if the pointlight doesn't exist, I'll play around with it.

**Comment by metalgearsloth** [comment]
Looks like some other PR fixed it.

**Pull request closed without merging**